### ISSUE TITLE
Proposal: Expirable LRU Disk Cache

### ISSUE THREAD

**Issue by dmytrodanylyk** [bumptech/glide#3418]
**Glide Version**: 4.8.0

With upcoming [GDPR](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation) it would be nice to have some disk cache expiration policy. E.g. I want to cache image for 30 days.

This can be achieved by modifying `DiskLruCacheWrapper#get` method.

```java
@Override
public File get(Key key) {
    // ...
    if(expired)  {
        result = null;
        delete(key);
    }
    // ...
}
```

If you think this may be useful I can send PR.

**Comment by dmytrodanylyk** [comment]
@sjudd @SUPERCILEXyz thoughts on this?

**Comment by sjudd** [comment]
Are you ok with leaving "expired" files sitting on disk until the next time someone happens to request them? 

If not, you probably need a periodic clean up task and some accounting of when the file expires. You might be able to write that into the DiskLruCache implementation Glide uses, but you'd probably  need to modify Glide's DiskCache interface to make this coherent, which is tricky.

**Comment by dmytrodanylyk** [comment]
> Are you ok with leaving "expired" files sitting on disk until the next time someone happens to request them?

@sjudd  I think that's ok for now. At least that's how I understand GDPR. Expired data must not be accessible.

**Comment by TWiStErRob** [comment]
If a Glide user relocates the cache to `/sdcard` then the file is accessible, so it doesn't cover all cases.

**Comment by dmytrodanylyk** [comment]
Yes sure, it doesn't cover all the edge cases. And I don't think it's possible to do that. At least with my proposal Glide will have the expirable caching mechanism out of the box.
